src/process-columnar-results.js

```diff
@@ -87,16 +87,17 @@ export default function processColumnarResults(
             case "TIME":
             case "TIMESTAMP":
             case "DATE":
-              row[fieldName].push(
-                data.columns[c].data.arr_col[r].data.int_col[e] *
-                  oneThousandMilliseconds
-              )
+              // See below for an explanation of these operations
+              // eslint-disable-next-line no-magic-numbers
+              const divisor = 10 ** (fieldPrecision - 3)
+              const timeInMs = data.columns[c].data.int_col[r] / divisor
+              row[fieldName].push(new Date(timeInMs))
```

Comment:
I just realized that the previous version of this code never converted the integer value to a JS `Date` object, it just converted it to milliseconds and left it as a Number. Should I revert back to this behavior? I assume so, just to match existing behavior, but would appreciate a second opinion.